libtracepoint-control-cpp/src/TracepointSession.cpp

```diff
@@ -74,107 +74,144 @@ RoundUpBufferSize(uint32_t pageSize, size_t bufferSize) noexcept
     return BufferSizeMax;
 }
 
-/*
-Manages a scatter-gather list of chunks of memory that need to be written to
-the perf.data file. This allows us to reduce the number of kernel calls.
-Instead of calling write() once per event (or twice if the event wraps), we
-call writev once for every 16 noncontinguous blocks of memory to be written.
-*/
-class IovecList
-{
-    static constexpr unsigned Max = 16;
-    unsigned m_used = 0;
-    iovec m_vecs[Max];
-
-public:
-
-    unsigned
-    RoomLeft() const noexcept
-    {
-        return Max - m_used;
-    }
+static tracepoint_decode::PerfEventSessionInfo
+MakeSessionInfo(uint32_t clockid) noexcept
```

Comment:
Note: Moved code out of the Session constructor so that the SessionInfo field can be const.